### DISCUSSION TITLE
API Feedback

### DISCUSSION THREAD

**Discussion by ernado** [ClickHouse/ch-go#6] [Announcements]
## 👋 Feedback is required

Most protocol and low-level parts are completed, so feel free to post higher-level API suggestions.

So, some points from myself:

1. Connection pooling (in roadmap)
2. ~~Column inferring (type is parsed during query execution)~~
3. ~~Automatic result inferring (names and column count is inferred during query execution), will require API change~~
4. Helpers for Arrays, LowCardinalitty, etc (in roadmap)

## Roadmap

* Alpha in Q1 22
* Beta in Q2 22
* Fully stable in Q3 22

**Comment by corpix** [comment]
Any serious reason not to use interface-based polymorphism and use generics instead?
Say we writing a data mapping software and have a table schema known at runtime while doing a query construction with current API... it will require tons of switch-case for complex types (like ColArray[T]) because generics are monomorphised at compile time.
